core/src/main/java/org/elasticsearch/search/suggest/completion/context/CategoryContextMapping.java

```diff
@@ -138,8 +142,11 @@ protected XContentBuilder toInnerXContent(XContentBuilder builder, Params params
             for (IndexableField field : fields) {
                 if (field.fieldType() instanceof KeywordFieldMapper.KeywordFieldType) {
                     values.add(field.binaryValue().utf8ToString());
-                } else {
+                } else if (field.fieldType() instanceof StringFieldType) {
                     values.add(field.stringValue());
+                } else {
+                    // ignore doc_values field
+                    assert field.fieldType() instanceof SortedDocValuesField || field.fieldType() instanceof SortedSetDocValuesField;
```

Comment:
Can we make it a real exception? As well as fail if none of the fields provided a value so that we fail on numeric fields, or unindexed keyword fields?